### PR TITLE
[stable/redis] add staticID feature to allow constant sentinel ids

### PR DESCRIPTION
**Pull request by rsecob**
#### What this PR does / why we need it:
We just deployed the cluster mode with sentinel for redis with 2 slaves and one master. After one pod failure the sentinels were able to elect a new master, after a second pod failure no master could be elected for the following reason:

> Sentinels never forget about replicas of a given master, even when they are unreachable for a long time. This is useful, because Sentinels should be able to correctly reconfigure a returning replica after a network partition or a failure event.
[Source](https://redis.io/topics/sentinel#removing-the-old-master-or-unreachable-replicas)

In Kubernetes, unless specified, each pod is given a random IP. In our case, with 2 slaves and one master, each replicas is attributed to a random IP. Holding on to old replicas IPs will only make it impossible for other sentinels to elect a new master. Although it is possible for a client to send the command `SENTINEL RESET *` to force sentinel servers to reset its list of sentinel peers. We can also tackle this issue by giving a constant sentinel ID which will allow other sentinel peers to update to the new given IP.

Currently, each sentinel is generating a new random ID on startup. The feature was developed to allow users to choose one way or the other.

#### Special notes for your reviewer:
For those who are familiar with the Redis chart, would there be another way to tackle this issue?

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by rsecob** [comment]
@juan131 @miguelaeh 
did you guys have a chance to look at this?

File: stable/redis/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: redis
-version: 10.0.1
+version: 10.0.2
```
Comment (javsalgar) [inline stable/redis/Chart.yaml]: I would bump the minor version, as it is an extra feature

**Comment by javsalgar** [comment]
Hi,

Thanks for the PR. I looked at it and LGTM. Could you bump the minor version?

File: stable/redis/Chart.yaml (hunk 1 above)
Comment (rsecob) [reply stable/redis/Chart.yaml]: Done :D

**Comment by rsecob** [comment]
@javsalgar thank you for checking this PR, I've applied the requested changes

**Comment by rsecob** [comment]
@javsalgar @juan131 @miguelaeh 
Tests have been rerun and passed, can one of you merge asap before we have another merge conflict haha, thank you :D

**Comment by dntosas** [comment]
@rsecob great work man ! :+1: :100:

**Pull request merged**